staging/src/k8s.io/client-go/testing/fake.go

```diff
@@ -131,13 +131,15 @@ func (c *Fake) Invokes(action Action, defaultReturnObj runtime.Object) (runtime.
 	c.Lock()
 	defer c.Unlock()
 
-	c.actions = append(c.actions, action.DeepCopy())
+	actionCopy := action.DeepCopy()
+	c.actions = append(c.actions, actionCopy)
```

Comment:
I'd continue to append a distinct copy to `c.actions` to capture the submitted object prior to any reactor modification, and just pass actionCopy into all the React calls below